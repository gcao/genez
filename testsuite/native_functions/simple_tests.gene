#!/usr/bin/env gene run

# Test: Simple native function tests

# Basic math
(print "sin(0) = " (sin 0))
(print "cos(0) = " (cos 0))
(print "sqrt(4) = " (sqrt 4))

# Type conversions
(print "int(3.14) = " (int 3.14))
(print "float(5) = " (float 5))
(print "str(42) = " (str 42))

# Collections
(print "len([1 2 3]) = " (len [1 2 3]))
(print "len(\"hello\") = " (len "hello"))

# Concatenation
(print "[1 2] ++ [3 4] = " (++ [1 2] [3 4]))
(print "\"hi\" ++ \"!\" = " (++ "hi" "!"))