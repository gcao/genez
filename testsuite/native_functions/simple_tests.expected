sin(0) =  0
cos(0) =  1
sqrt(4) =  2
int(3.14) =  3
float(5) =  5
str(42) =  42
len([1 2 3]) =  3
len(\"hello\") =  5
[1 2] ++ [3 4] =  [1 2 3 4]
\"hi\" ++ \"!\" =  hi!